apps/src/code-studio/peer_reviews/PeerReviewSubmissions.jsx

```diff
@@ -1,41 +1,45 @@
 import React from 'react';
-import {FormControl} from 'react-bootstrap';
+import {Button, FormControl} from 'react-bootstrap';
 import PropTypes from 'prop-types';
 import _ from 'lodash';
 import Spinner from '../pd/workshop_dashboard/components/spinner';
 import PeerReviewSubmissionData from "./PeerReviewSubmissionData";
+import $ from 'jquery';
 
 class PeerReviewSubmissions extends React.Component {
   static propTypes = {
-    filterType: PropTypes.string.isRequired
+    filterType: PropTypes.string.isRequired,
+    courseList: PropTypes.arrayOf(PropTypes.array).isRequired
   }
 
   state = {}
 
   componentWillMount() {
     this.getFilteredResults = _.debounce(this.getFilteredResults, 1000);
 
-    this.loadRequest = $.ajax({
-      method: 'GET',
-      url: `/api/v1/peer_review_submissions/index?filter=${this.props.filterType}`,
-      dataType: 'json'
-    }).done(data => {
-      this.setState({
-        submissions: data
-      });
+    this.getFilteredResults();
+  }
+
+  handleCourseFilterChange = (event) => {
+    this.setState({plc_course_id: event.target.value}, () => {
+      this.getFilteredResults();
```

Comment:
I don't think we need to wait for the `setState` to complete before making the ajax call. I'd prefer to get the filtered results immediately. Instead of relying on `this.state.plc_course_id` on line 42 below, modify `getFilteredResults` to take that id as an argument and make the call immediately.

As a bonus you could add a `loading` bool or similar state and display a spinner while it's loading for better user feedback, like the [workshop table loader](https://github.com/code-dot-org/code-dot-org/blob/8475c51e3981dc8e1d5e554a1f2e1bfded9cd253/apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx#L95-L102).